chalk-solve/src/clauses/builtin_traits/sized.rs

```diff
@@ -74,9 +74,11 @@ pub fn add_sized_program_clauses<I: Interner>(
             TypeName::Tuple(arity) => {
                 push_tuple_sized_conditions(db, builder, trait_ref, *arity, substitution)
             }
-            TypeName::Never | TypeName::Scalar(_) | TypeName::Raw(_) | TypeName::Ref(_) => {
-                builder.push_fact(trait_ref.clone())
-            }
+            TypeName::Array
```

Comment:
Do we have "WF" condition code somewhere? We'll have to check that the `[T;N]` requires `T: Sized`

Reply:
We push WF clauses in `match_type_name`, and I don't think we had a big discussion about how we want to handle well-formedness of tuples and arrays. IIRC the takeaway was that we rely on rustc to provide well-formed tuples (and, in this case, arrays), but I might be wrong

Reply:
I guess I was thinking of [`WfSolver`](http://rust-lang.github.io/chalk/chalk_solve/wf/struct.WfSolver.html) -- but yeah I think we probably shouldn't block on this right now, it's something we'll definitely want to think about.

Reply:
We should just file an issue I guess.